website/content/en/status/report-2023-01-2023-03/drm-drivers.adoc

```diff
@@ -0,0 +1,45 @@
+=== DRM drivers (i.e. GPU drivers)
+
+Links: +
+link:https://github.com/freebsd/drm-kmod[Git repository on GitHub] URL: link:https://github.com/freebsd/drm-kmod[https://github.com/freebsd/drm-kmod] +
+
+Contact: Emmanuel Vadot <manu@FreeBSD.org> +
+Contact: Jean-Sébastien Pédron <dumbbell@FreeBSD.org>
+
+GPUs are driven by DRM drivers.
+They are developped specifically for Linux using a permissive license.
+Our mission is to port those drivers to FreeBSD to make sure modern GPUs are fully supported.
+
+We didn't publish a report to share our progress for a long time.
+Therefore this status report entry will cover more than just the last quarter.
+
+==== Update to Linux 5.15 LTS and Linux 5.16
+
+As of this status report, the `drm-kmod` meta port still installs the DRM drivers from Linux 5.10 (released on December 13, 2020).
+This version lacks support for recent GPUs, in particular Intel 12th gen Alder Lake ones.
+In the past months, we worked to update the DRM drivers to bring support for more modern AMD and Intel GPUs.
+
+The `drm-kmod` Git repository `master` branch was first updated to Linux 5.15 (released on October 31, 2021).
+This is an LTS branch in Linux and we wanted to take advantage of that.
+Thus at that point, we followed two paths:
+* A `5.15-lts` was created to backport all bug fixes from Linux 5.15.x patch releases. This work is now available in the `drm-515-kmod` port.
+* The porting effort from subsequence Linux versions continued. The `master` branch is now at Linux 5.16 (release on January 9, 2022).
+
+The Intel driver from Linux 5.15 LTS supports 12th gen GPUs (Alder Lake).
+We still need more testing and a few fixes (which are ready otherwise).
+That's why the `drm-kmod` still installs `drm-510-kmod` by default instead of `drm-515-kmod`.
+But at last, FreeBSD should run as a desktop on this GPU generation.
+
+==== Linux 5.17 and future work
+
+DRM drivers from Linux 5.17 (released on March 20, 2022) were already ported but link:https://github.com/freebsd/drm-kmod/pull/236[this work still sits in its own branch].
+
+There are a couple issues which block further testing and the merge into the `master` branch:
+* Our current integration with vt(4), the console/terminal driver, is quite far from the DRM drivers expectations which  are based on Linux' fbdev KPI. Something changed in both the Intel and AMD drivers, meaning that vt(4) breaks with the 5.17 update.
+* The initial Linux 5.17 release does not contain the fixes backported to Linux 5.15 LTS. It seems quite unstable with the Intel 12th gen GPU mentionned earlier.
+
+To address the issue with our vt(4) integration layer, we started to link:https://github.com/freebsd/drm-kmod/pull/243[write a new vt(4) backend specifically to use the fbdev callbacks exposed by the DRM drivers].
```

Comment:
s/our vt(4)/our man:vt[4]/ (but not the 2nd vt(4) otherwise you'd end with <a> embedded inside <a>)

Reply:
Fixed, thanks!

Reply:
Just to be clear, I changed one instace of "vt(4)" to "man:vt[4]". Is that correct?